src/xia2/Modules/Indexer/DialsIndexer.py

```diff
@@ -346,6 +355,50 @@ def _index_prepare(self):
                 raise RuntimeError("No spots found in sweep %s" % xsweep.get_name())
             logger.info(spot_counts_per_image_plot(refl))
 
+            # in terms of checking for saturation; measure (i) the maximum pixel
+            # in each spot then also a histogram of the totals to get a sense of
+            # _how_ overloaded things are
+
+            if PhilIndex.params.general.check_for_saturated_pixels:
+                t0 = time.time()
+                count_limit = imageset.get_detector()[0].get_trusted_range()[1]
+                maxima = flex.double()
+                boxes = refl["shoebox"]
+                nn = boxes.size()
+
+                pixel_histogram = flex.histogram(
+                    flex.double(), data_min=0, data_max=1.25 * count_limit, n_slots=5
+                )
+
+                for j in range(nn):
+                    b = boxes[j]
+                    pixel_histogram.update(
+                        flex.histogram(
+                            b.data.as_double().as_1d(),
+                            data_min=0,
+                            data_max=1.25 * count_limit,
+                            n_slots=5,
+                        )
+                    )
```

Comment:
Is this quicker than concatenating all the flattened shoebox pixel intensity values and simply calling `flex.histogram` (or `numpy.bincount` or whatever) on the single array?

Reply:
My concern was more memory footprint rather than calculation time - though I have no data to assess this. 

I'll run with a big data set & record timing for comparison.

Reply:
Now running with a 9,000 image semi-saturated Eiger 16M data set to get worst-case timing added with current protocol.

Reply:
OK, for this case it took 

```
Overloads found: 5 spots / 5 pixels
Overload detection took 3.32s
```

which I think is probably OK given 

```
675716 spots found on 9000 images (max 11941 / bin)
```